### PR TITLE
Fix the ag-grid theme menu

### LINKED ISSUE #1222: uitk-ag-theme uses same padding in .ag-menu for all densities
### Package name(s)

AG Grid Theme v1.0.0 (@salt-ds/ag-grid-theme)

### Problem
The theme applies the same padding to .ag-menu for all densities. This seems okay for Medium and Low densities, but it looks too big for HD.
I should probably scale accordingly for each of the different densities.

This value is defined here:
https://github.com/jpmorganchase/salt-ds/blob/main/packages/ag-grid-theme/css/_uitk-ag-theme-mixin.scss#L94

#### Low density
image

#### Medium density
image

#### High density
(Please ignore the fact that the dropdown looks bigger for HD, there is a fix for that already)
image

### Steps to reproduce

1. Visit [ag-theme story](https://storybook.saltdesignsystem.com/?path=/docs/documentation-data-grid-ag-grid-theme--page)
2. Scroll to "floating filter" story
3. Switch the theme to "UITK"
4. Click on the filter icon.
5. Repeat for the different densities

### Expected behavior

- Padding for menu scales based on density

### Operating system

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by ZarkoPernar**
Closes #1222

### CODE REVIEW

File: packages/ag-grid-theme/css/uitk-ag-theme.scss [hunk 1]
```diff
@@ -1,9 +1,27 @@
 @import "./uitk-ag-theme-mixin";
 
+div[class*="ag-theme-uitk-high"] {
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/uitk-ag-theme.scss:3]: should these not be --salt? + if so, the last one should then be container-spacing not container-padding

Comment (ZarkoPernar) [reply packages/ag-grid-theme/css/uitk-ag-theme.scss:3]: This is for uitk theme specifically

Comment (ZarkoPernar) [reply packages/ag-grid-theme/css/uitk-ag-theme.scss:3]: Ok, can do

**Review by libertymayc — DISMISSED** [review]

**Review by libertymayc — APPROVED** [review]

**Comment by ZarkoPernar** [comment]
@joshwooding waiting or review from @dplsek

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 2]
```diff
@@ -77,10 +77,59 @@
 
   .ag-menu {
     background-color: var(--agGrid-background);
-    padding: var(--salt-size-unit) var(--salt-size-unit) 0 var(--salt-size-unit);
+    padding: calc(var(--salt-size-unit) * 2);
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:80]:
```suggestion
    padding: var(--salt-spacing-200);
```

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 3]
```diff
@@ -77,10 +77,59 @@
...
+    padding: 0;
+    padding-top: var(--salt-size-unit);
+  }
+
+  // start filter input
+  .ag-menu .ag-filter-body-wrapper {
+    padding: 0;
+    display: flex;
+    flex-direction: column;
+    gap: var(--salt-size-unit);
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:94]:
```suggestion
    gap: var(--salt-spacing-100);
```

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 4]
```diff
@@ -77,10 +77,59 @@
...
+    border-radius: 0;
+  }
+
+  .ag-picker-field-icon {
+    margin-right: var(--salt-size-unit);
+  }
+
+  .ag-select .ag-picker-field-display {
+    margin: 0;
+    padding-left: var(--salt-size-unit);
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:112]:
```suggestion
    padding-left: var(--salt-spacing-100);
```

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 5]
```diff
@@ -77,10 +77,59 @@
...
+  }
+
+  .ag-menu .ag-input-field-input[type="text"] {
+    height: var(--agGrid-input-height);
+    min-height: var(--agGrid-input-height);
+    line-height: var(--agGrid-input-height);
+    padding: 0 var(--salt-size-unit);
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:119]:
```suggestion
    padding: 0 var(--salt-spacing-100);
```

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 6]
```diff
@@ -103,6 +152,10 @@
     height: var(--salt-size-stackable);
   }
 
+  .ag-tab:only-child {
+    margin: 0;
+  }
+
   .ag-tab-selected {
     border-bottom: solid 2px var(--agGrid-tab-borderColor-active);
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:160]:
```suggestion
    border-bottom: var(--salt-editable-borderStyle) var(--salt-editable-borderWidth-active) var(--agGrid-tab-borderColor-active);
```

Comment (libertymayc) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:160]:
```suggestion
    border-bottom: var(--salt-editable-borderStyle) var(--salt-editable-borderWidth-active) var(--agGrid-tab-borderColor-active);
```

File: packages/ag-grid-theme/css/_uitk-ag-theme-mixin.scss [hunk 7]
```diff
@@ -91,10 +91,42 @@
...
+    padding: 0;
+    padding-top: 8px;
+  }
+
+  // start filter input
+  .ag-menu .ag-filter-body-wrapper {
+    padding: 0;
+    display: flex;
+    flex-direction: column;
+    gap: 8px;
```
Comment (libertymayc) [inline packages/ag-grid-theme/css/_uitk-ag-theme-mixin.scss:108]: fixed at 8px?

File: packages/ag-grid-theme/css/uitk-ag-theme.scss (hunk 1 above)
Comment (libertymayc) [reply packages/ag-grid-theme/css/uitk-ag-theme.scss:3]: I wonder if you want to align these with the new spacing tokens
would be 
--uitk-spacing-100 for uitk-size-unit
--uitk-spacing-200 for uitk-container-spacing

**Comment by ZarkoPernar** [comment]
@libertymayc implemented all of your suggestions [here](https://github.com/jpmorganchase/salt-ds/pull/1500/commits/10be491ea82af36ce54e773761805cf4dc5701a4) That was really helpful, thank 👍

**Review by joshwooding — APPROVED** [review]

**Pull request merged**